Sources/SwiftProtobuf/Message+AsyncMessageSequence.swift

```diff
@@ -0,0 +1,232 @@
+//
+// Sources/SwiftProtobuf/AsyncMessageSequence.swift - Async sequence over binary delimited protobuf
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// An async sequence of messages decoded from a binary delimited protobuf stream.
+///
+// -----------------------------------------------------------------------------
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+extension Message {
+  
+  /// Creates an asynchronous sequence of size-delimited messages from this sequence of bytes.
+  /// Delimited format allows a single file or stream to contain multiple messages. A delimited message
+  /// is a varint encoding the message size followed by a message of exactly that size.
+  ///
+  /// - Parameters:
+  ///   - baseSequence: The `AsyncSequence` to read messages from.
+  ///   - extensions: An `ExtensionMap` used to look up and decode any extensions in
+  ///    messages encoded by this sequence, or in messages nested within these messages.
+  ///   - partial: If `false` (the default),  after decoding a message, `Message.isInitialized`
+  ///     will be checked to ensure all fields are present. If any are missing,
+  ///     `BinaryEncodingError.missingRequiredFields` will be thrown.
+  ///   - options: The BinaryDecodingOptions to use.
+  /// - Returns: An asynchronous sequence of messages read from the `AsyncSequence` of bytes.
+  /// - Throws: `BinaryDecodingError` if decoding fails, throws
+  ///           `BinaryDelimited.Error` for some reading errors,
+  ///           `BinaryDecodingError.truncated` if the sequence ends before fully decoding a
+  ///           message or a delimiter,
+  ///           `BinaryDecodingError.malformedProtobuf`if a delimiter could not be read and
+  ///           `BinaryDecodingError.tooLarge` if a size delimiter of 2GB or greater is found.
+  ///
+public static func asyncSequence<Base: AsyncSequence> (
+    baseSequence: Base,
+    extensions: ExtensionMap? = nil,
+    partial: Bool = false,
+    options: BinaryDecodingOptions = BinaryDecodingOptions()
+  ) -> AsyncMessageSequence<Base, Self> {
+    AsyncMessageSequence<Base, Self>(
+      baseSequence: baseSequence,
+      extensions: extensions,
+      partial: partial,
+      options: options
+    )
+  }
+}
```

Comment:
Let's remove this extension. The usability here is really questionable in my opinion. Your normally start from an async sequence and then apply transformations and that is how the whole ecosystem has been shaped up.

Reply:
Yes done. Renamed file to AsyncMessageSeqeunce.swift because Message+AsyncMessageSequence.swift didn't make sense anymore.